### PR TITLE
Add a way to run all plugin test from a rake command

### PR DESCRIPTION
**Pull request by purbon**
Added a rake task to run all test plus fix  dependency for octokit.

### CODE REVIEW

File: rakelib/test.rake [hunk 1]
```diff
@@ -16,6 +16,11 @@ namespace "test" do
     RSpec::Core::Runner.run(["--fail-fast", *Rake::FileList["spec/**/*.rb"]])
   end
 
+  task "plugins" => [ "bootstrap","plugin:install-all" ] do
+    pattern = "vendor/bundle/jruby/1.9/gems/logstash-*/spec/{input,filter,codec,output}s/*_spec.rb"
```
Comment (ph) [inline rakelib/test.rake]: Can we use the `logstash::Environment.logstash_gem_home`? to generare the path?

Comment (purbon) [inline rakelib/test.rake]: again :-(, thanks for the point.

File: rakelib/test.rake [hunk 2]
```diff
@@ -16,6 +16,11 @@ namespace "test" do
     RSpec::Core::Runner.run(["--fail-fast", *Rake::FileList["spec/**/*.rb"]])
   end
 
+  task "plugins" => [ "bootstrap","plugin:install-all" ] do
```
Comment (suyograo) [inline rakelib/test.rake]: can we call this test_plugins or something more explicit?

File: rakelib/test.rake [hunk 3]
```diff
@@ -16,6 +16,11 @@ namespace "test" do
...
+    pattern = "vendor/bundle/jruby/1.9/gems/logstash-*/spec/{input,filter,codec,output}s/*_spec.rb"
+    sh "./bin/logstash rspec #{pattern}"
```
Comment (ph) [inline rakelib/test.rake]: This call require us to be at the root of logstash, we should use the `Logstash::LOGSTASH_HOME` constant to generate the path so we can run this task anywhere.

File: rakelib/test.rake (hunk 2 above)
Comment (purbon) [inline rakelib/test.rake]: it is under test, so to call this you have to do ```rake test:plugins``

File: rakelib/test.rake [hunk 4]
```diff
@@ -16,6 +16,13 @@ namespace "test" do
     RSpec::Core::Runner.run(["--fail-fast", *Rake::FileList["spec/**/*.rb"]])
   end
 
+  task "plugins" => [ "bootstrap","plugin:install-all" ] do
+    require "logstash/environment"
+    gem_home = LogStash::Environment.logstash_gem_home
+    pattern = "#{gem_home}/logstash-*/spec/{input,filter,codec,output}s/*_spec.rb"
+    sh "./bin/logstash rspec #{pattern}"
+  end
```
Comment (ph) [inline rakelib/test.rake]: This call require us to be at the root of logstash, we should use the Logstash::LOGSTASH_HOME constant to generate the path so we can run this task anywhere.

File: rakelib/test.rake (hunk 2 above)
Comment (suyograo) [inline rakelib/test.rake]: ah didn't notice. thanks :+1:

Comment (suyograo) [inline rakelib/test.rake]: actually how about all plugins?

Comment (purbon) [inline rakelib/test.rake]: it test all plugins, could you elaborate this a bit more?

Comment (ph) [inline rakelib/test.rake]: `rake test:all-plugins` @suyograo is that what you mean?

Comment (purbon) [inline rakelib/test.rake]: I can add this, makes sense @ph

Comment (suyograo) [inline rakelib/test.rake]: +1, thats what I meant :)

Comment (purbon) [inline rakelib/test.rake]: done

File: rakelib/test.rake (hunk 4 above)
Comment (purbon) [inline rakelib/test.rake]: the way logstash_gem_home is generated already does this.

Comment (ph) [inline rakelib/test.rake]: Yes but not for our test wrapper, ```sh "./bin/logstash"```

Comment (purbon) [inline rakelib/test.rake]: auch, stupid me, sorry.

File: rakelib/test.rake [hunk 5]
```diff
@@ -16,6 +16,13 @@ namespace "test" do
     RSpec::Core::Runner.run(["--fail-fast", *Rake::FileList["spec/**/*.rb"]])
   end
 
+  task "all-plugins" => [ "bootstrap","plugin:install-all" ] do
+    require "logstash/environment"
+    gem_home = LogStash::Environment.logstash_gem_home
+    pattern = "#{gem_home}/logstash-*/spec/{input,filter,codec,output}s/*_spec.rb"
+    sh "#{LogStath::Environment::LOGSTASH_HOME}/bin/logstash rspec #{pattern}"
```
Comment (electrical) [inline rakelib/test.rake]: please add --order rand to it.

Comment (purbon) [inline rakelib/test.rake]: is this not run by default by devutils?

Comment (electrical) [inline rakelib/test.rake]: When supplying any arguments it removes all defaults.
Also devutils is not in use for this part.

**Comment by electrical** [comment]
LGTM

**Pull request closed without merging**